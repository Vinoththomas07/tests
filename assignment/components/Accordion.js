import React, { Component } from "react";
import { Text, View, TouchableOpacity, StyleSheet } from "react-native";

export default class Accordion extends Component {
  constructor(props) {
    super(props);
    this.state = {
      timer: 0,
      showDetails: "none",
    };
  }

  renderValue(value, title, symbol) {
    return (
      <View style={{ alignItems: "center" }}>
        <Text style={{ fontSize: 12 }}>{title}</Text>
        {value ? (
          <Text style={{ fontSize: 12, color: "#828282" }}>
            {symbol}
            {Math.round(value)}
          </Text>
        ) : (
          <Text style={{ fontSize: 12, color: "#828282" }}>No data</Text>
        )}
      </View>
    );
  }

  render() {
    const { item } = this.props;
    const { showDetails } = this.state;
    return (
      <View style={styles.container}>
        <TouchableOpacity
          style={styles.listItem}
          onPress={() => {
            showDetails == "flex"
              ? this.setState({ showDetails: "none" })
              : this.setState({ showDetails: "flex" });
          }}
        >
          <View style={styles.header}>
            <Text style={styles.headerLeft}>{item.name}</Text>
            <View>
              <Text>₹{Math.round(item.quote.INR.price)}</Text>
            </View>
          </View>
          <View style={styles.body,{display:showDetails}}>
            <View style={styles.bodyRow}>
              {this.renderValue(item.quote.INR.market_cap, "Market Cap", "₹")}
              {this.renderValue(item.quote.INR.volume_24h, "Volume", "₹")}
            </View>
            <View style={styles.bodyRow}>
              {this.renderValue(item.max_supply, "Max Supply")}
              {this.renderValue(item.total_supply, "Total Supply")}
            </View>
          </View>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    marginTop: 20,
  },
  listItem: {
    display: "flex",
    elevation: 1,
    borderTopLeftRadius: 15,
    borderTopRightRadius: 15,
    borderBottomRightRadius: 15,
    borderBottomLeftRadius: 15,
    backgroundColor: "#fff",
  },
  header: {
    flexDirection: "row",
    justifyContent: "space-between",
    padding: 25,
    paddingHorizontal: 30,
  },
  body: {
    paddingBottom: 25,
    paddingHorizontal: 30,
  },
  headerLeft: {
    fontWeight: "bold",
    fontSize: 16,
  },
  bodyRow: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-around",
    paddingBottom: 20,
  },
});
