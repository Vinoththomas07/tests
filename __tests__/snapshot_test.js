import React from 'react';
import {shallow} from 'enzyme';
import Task1 from '../assignment/task1';
import Task2a from '../assignment/task2a';
import Task2b from '../assignment/task2b';
import Accordion from '../assignment/components/Accordion'

describe('Snapshot test', () => {
    describe('Rendering', () => {
        it('Task1, should match to snapshot', () => {
            const component = shallow(<Task1/>)
            expect(component).toMatchSnapshot()
        });
        it('Task2a, should match to snapshot', () => {
            const component = shallow(<Task2a/>)
            expect(component).toMatchSnapshot()
        });
        it('Task2b, should match to snapshot', () => {
            const component = shallow(<Task2b/>)
            expect(component).toMatchSnapshot()
        });
    });
});