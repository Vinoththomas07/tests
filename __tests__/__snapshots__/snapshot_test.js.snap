// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Rendering Task1, should match to snapshot 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "justifyContent": "center",
      "margin": 20,
    }
  }
>
  <Text
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    You should see the pyramid text displayed in the console, search for DEXHIGH in logs
  </Text>
  <Styled(Button)
    onPress={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "borderRadius": 30,
        "margin": 15,
        "padding": 15,
      }
    }
    success={true}
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
       See next Task 
    </Text>
  </Styled(Button)>
</Styled(ViewNB)>
`;

exports[`Snapshot test Rendering Task2a, should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "alignSelf": "center",
      "display": "flex",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginVertical": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      Timer in seconds
    </Text>
    <Text
      style={
        Object {
          "fontSize": 80,
          "fontWeight": "bold",
        }
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Styled(Button)
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "padding": 10,
          "width": 120,
        }
      }
      success={true}
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
         Start
      </Text>
    </Styled(Button)>
    <Styled(Button)
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "padding": 10,
          "width": 120,
        }
      }
      warning={true}
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
         Stop and Reset
      </Text>
    </Styled(Button)>
  </View>
</View>
`;

exports[`Snapshot test Rendering Task2b, should match to snapshot 1`] = `
<index
  backgroundColor="#4B4BC9"
  bounces={true}
  foreground={[Function]}
  headerType="AvatarHeader"
  image={
    Object {
      "testUri": "../../../assets/linker.png",
    }
  }
  leftTopIcon={null}
  parallaxHeight={250}
  renderBody={[Function]}
  rightTopIcon={null}
  subtitle="Top 100 Crypto currency list"
  title="Linkercoin"
/>
`;
